src/parser/HLSTree.cpp

```diff
@@ -284,12 +284,22 @@ bool adaptive::CHLSTree::ParseChildManifest(const std::string& data,
 
       switch (ProcessEncryption(rep->GetBaseUrl(), attribs))
       {
+        case EncryptionType::CLEAR:
+          currentEncryptionType = EncryptionType::CLEAR;
+          period->SetEncryptionState(EncryptionState::UNENCRYPTED);
+          psshSetPos = PSSHSET_POS_DEFAULT;
+          break;
         case EncryptionType::UNKNOWN:
+          currentEncryptionType = EncryptionType::UNKNOWN;
+          period->SetEncryptionState(EncryptionState::ENCRYPTED);
+          break;
         case EncryptionType::NOT_SUPPORTED:
+          currentEncryptionType = EncryptionType::NOT_SUPPORTED;
           period->SetEncryptionState(EncryptionState::ENCRYPTED);
           break;
         case EncryptionType::AES128:
           currentEncryptionType = EncryptionType::AES128;
+          period->SetEncryptionState(EncryptionState::UNENCRYPTED);
```

Comment:
@glennguy using this state case "unencrypted" for AES, is not ambiguous to understand?
i think is so because of this line:

https://github.com/xbmc/inputstream.adaptive/blob/0bea32373b454d64782d83744af782eb6b576c30/src/parser/HLSTree.cpp#L683

could be appropriate add a new EncryptionState enum state? e.g. ENCRYPTED_SUPPORTED_SW / ENCRYPTED_SUPPORTED_INT or other names?

i can also keep as is for now, but i think could be appropriate to do in the future

Reply:
Way back several years ago I interpreted 'encrypted' in this context to mean DRM,/Widevine as this is what the code realy means. 
I agree, it doesn't make much sense but I was sticking with convention at the time. Let's aim to make it read better in future

Reply:
Or in other words, 'encrypted' was really trying to say 'we need to initialise DRM for this' and AES-128 wasn't really counted as encrypted